From 24a62bacdbd09a3a7f93ae6842c9cfb3460010df Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:05:14 -0400
Subject: [powerpc] ppc64: Remove the unneeded load of ti_flags in resume_kernel

Message-id: <1410545655-205645-86-git-send-email-dzickus@redhat.com>
Patchwork-id: 94158
O-Subject: [RHEL7 PATCH 085/626] powerpc/ppc64: Remove the unneeded load of ti_flags in resume_kernel
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 0edfdd10f57bd989f7c2bc31ce6f601bbee1b664
Author: Kevin Hao <haokexin@gmail.com>
Date:   Thu Sep 26 16:41:34 2013 +0800

    powerpc/ppc64: Remove the unneeded load of ti_flags in resume_kernel

    We already got the value of current_thread_info and ti_flags and store
    them into r9 and r4 respectively before jumping to resume_kernel. So
    there is no reason to reload them again.

    Signed-off-by: Kevin Hao <haokexin@gmail.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/entry_64.S b/arch/powerpc/kernel/entry_64.S
index f962e11..662c6dd 100644
--- a/arch/powerpc/kernel/entry_64.S
+++ b/arch/powerpc/kernel/entry_64.S
@@ -681,9 +681,7 @@ _GLOBAL(ret_from_except_lite)
 
 resume_kernel:
  /* check current_thread_info, _TIF_EMULATE_STACK_STORE */
- CURRENT_THREAD_INFO(r9, r1)
- ld r8,TI_FLAGS(r9)
- andis. r8,r8,_TIF_EMULATE_STACK_STORE@h
+ andis. r8,r4,_TIF_EMULATE_STACK_STORE@h
  beq+ 1f
 
  addi r8,r1,INT_FRAME_SIZE /* Get the kprobed function entry */
-- 
1.7.1